### ISSUE TITLE
Persistence

### ISSUE THREAD

**Issue by heathharrelson** [OCTRI/phenotypr-body#28]
## Description

Save the selected terms (symptom, HPO term, HPO#) for research analysis. The results do not need to be matched to a participant, rather an independent session ID. As searches are performed, results should be captured once they click the "Done adding symptoms" button to view the results page. 

## Acceptance Criteria

* HTTP request is performed on transition to the results route
* HTTP request contains the selected terms

## Dependencies

* Need to know the Phenotypr API endpoint to transmit the request to
* Need to provide spec of the request body to BCH
* Depending on implementation, Phenotypr API may need to handle cross-origin requests by setting the `Access-Control-Allow-Origin` header appropriately

**Comment by heathharrelson** [comment]
I'm imagining the request payload looking something like this:

```json
{
  "session_id": "a UUID",
  "selected_terms": [
    {
      "id": "HP:XXXXXXX",
      "label": "HPO Term Label from Solr",
      "symptom": "Plain language term selected by the patient"
    }
  ]
}
```

**Comment by heathharrelson** [comment]
Per discussion with Nitin on 06/22, here is the revised payload with scoring information:

```json
{
  "session_id": "a UUID",
  "selected_terms": [
    {
      "id": "HP:XXXXXXXX"
      "label": "HPO Term Label fro Solr",
      "symptom": "Plain language synonym selected by the patient"
    }
  ],
  "sufficiency_score": 0.42
}
```

Sufficiency score is a JavaScript `Number` type (double precision IEEE-754). It should be a real number between 0 and 1.

**Comment by heathharrelson** [comment]
As of PR #67, the persistence payload is shaped like this:

```json
{
  "session_id": "a UUID",
  "selected_systems": [
    "HP:0001507",
    "HP:0000818"
  ],
  "selected_terms": [
    {
      "id": "HP:XXXXXXXX",
      "label": "HPO Term Label fro Solr",
      "symptom": "Plain language synonym selected by the patient"
    },
    {
      "id": "HP:XXXXXXXY",
      "label": "Second HPO Term Label from Solr",
      "symptom": "Another synonym"
    }
  ],
  "constrained_terms": [
    {
      "id": "HP:XXXXXXXX",
      "label": "HPO Term Label fro Solr",
      "symptom": "Plain language synonym selected by the patient"
    }
  ],
  "unconstrained_terms": [
    {
      "id": "HP:XXXXXXXY",
      "label": "Second HPO Term Label from Solr",
      "symptom": "Another synonym"
    }
  ]
  "found_all": true,
  "sufficiency_score": 0.42
}
```

The new fields are:

* `selected_systems`: HPO IDs of high-level systems selected by the user
* `constrained_terms`: An array of HPO terms (same structure as `selected_terms`) selected when search was constrained by high-level systems
* `unconstrained_terms`: An array of HPO terms (same structure as `selected_terms`) selected when the search was optionally constrained
* `found_all`: A boolean indicating whether the user was able to find all the conditions they wanted

@nitingujralBCH If this structure is problematic for BCH, it can easily be reworked.

**Comment by kirenotneb** [comment]
Resolved by #76
